ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/StructMemberHelper.java

```diff
@@ -0,0 +1,203 @@
+package com.spectralogic.ds3autogen.c.helpers;
+
+import com.google.common.collect.ImmutableList;
+import com.spectralogic.ds3autogen.c.models.StructMember;
+import com.spectralogic.ds3autogen.utils.Helper;
+import com.spectralogic.ds3autogen.utils.collections.GuavaCollectors;
+
+import java.text.ParseException;
+
+import static com.spectralogic.ds3autogen.utils.Helper.indent;
+
+public class StructMemberHelper {
+    private StructMemberHelper() {}
+
+    private final static StructMemberHelper structMemberHelper = new StructMemberHelper();
+
+    public static StructMemberHelper getInstance() {
+        return structMemberHelper;
+    }
+
+
+    public static ImmutableList<StructMember> getWrappedListChildNodes(final ImmutableList<StructMember> structMembers) {
+        return structMembers.stream()
+                .filter(sm -> sm.getType().isArray())
+                .filter(StructMember::hasWrapper)
+                .collect(GuavaCollectors.immutableList());
+    }
+
+    /**
+     * Used by ResponseParse.ftl and ResponseParserTopLevel.ftl
+     */
+    public static ImmutableList<StructMember> getUnwrappedListChildNodes(final ImmutableList<StructMember> structMembers) {
+        return structMembers.stream()
+                .filter(sm -> sm.getType().isArray())
+                .filter(sm -> !sm.hasWrapper())
+                .collect(GuavaCollectors.immutableList());
+    }
+
+    public static String generateStructMemberParserLine(final StructMember structMember, final String parserFunction) throws ParseException {
+        return indent(3) + "response->" + Helper.camelToUnderscore(structMember.getName()) + " = " + parserFunction + "\n";
+    }
+
+    public static String generateStructMemberDs3StrArrayBlock(final StructMember structMember) {
+        return indent(3) + "xmlNodePtr loop_node;\n"
```

Comment:
That's how the code lines up with the template that calls it (/ds3_autogen/ds3-autogen-c/src/main/resources/templates/source-templates/ResponseParser.ftl) ... I want to move these to small templates at some point very soon but have been working through higher priority problems